gtsfm/densify/patchmatchnet_data.py

```diff
@@ -0,0 +1,267 @@
+"""Interface class from GtsfmData to PatchmatchNetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.common.image import Image
+from gtsfm.densify.mvs_math import piecewise_gaussian
+
+NUM_PATCHMATCHNET_STAGES = 4
+
+MIN_DEPTH_PERCENTILE = 1
+MAX_DEPTH_PERCENTILE = 99
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for PatchmatchNet. It contains the interface from GtsfmData.
+    Ref: Wang et al. https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/datasets/dtu_yao_eval.py
+    """
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Cache image and camera pose metadata for PatchmatchNet inference.
+
+        Args:
+            images: input images (H, W, C) to GTSFM
+            sfm_result: sfm results calculated by GTSFM
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
+        """
+        assert images is not None and len(images) > 1
+
+        # Cache sfm result
+        self._sfm_result = sfm_result
+
+        # PatchmatchNet meta
+        self._num_views = num_views
+        self._num_stages = NUM_PATCHMATCHNET_STAGES
+
+        # Test data preparation
+        self._patchmatchnet_idx_to_camera_idx = sorted(self._sfm_result.get_valid_camera_indices())
+        self._num_images = len(self._patchmatchnet_idx_to_camera_idx)
```

Comment:
Might be nice to leave an inline comment explaining that this is not the number of images provided to GTSFM, but simply the number of images with estimated poses

Reply:
maybe we should name it something like `self._num_images_cal` for number of images that are calibrated (from calibrated cameras)